### PR TITLE
Part H of transition to support modernize-use-default-member-init

### PR DESCRIPTION
**Pull request by gnawme**
Part H of #5665, the transition to add these checks to `clang-tidy`:

- [modernize-use-default-member-init](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-default-member-init.html)

- [modernize-use-bool-literals](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-bool-literals.html)

### CODE REVIEW

File: surface/include/pcl/surface/gp3.h [hunk 1]
```diff
@@ -288,28 +274,28 @@ namespace pcl
...
-      double search_radius_;
+      double search_radius_{0.0};
 
       /** \brief The maximum number of nearest neighbors accepted by searching. */
-      int nnn_;
+      int nnn_{100};
 
       /** \brief The preferred minimum angle for the triangles. */
-      double minimum_angle_;
+      double minimum_angle_{0.0};
```
Comment (larshg) [inline surface/include/pcl/surface/gp3.h:286]:
```suggestion
      double minimum_angle_{M_PI/18};
```

As per constructor - applies to multiple parameters.

File: surface/include/pcl/surface/convex_hull.h [hunk 2]
```diff
@@ -88,12 +88,13 @@ namespace pcl
...
 
       /** \brief Empty constructor. */
-      ConvexHull () : compute_area_ (false), total_area_ (0), total_volume_ (0), dimension_ (0),
-                      projection_angle_thresh_ (std::cos (0.174532925) ), qhull_flags ("qhull "),
-                      x_axis_ (1.0, 0.0, 0.0), y_axis_ (0.0, 1.0, 0.0), z_axis_ (0.0, 0.0, 1.0)
-      {
-      }
-      
+      ConvexHull()
+      : qhull_flags("qhull ")
```
Comment (larshg) [inline surface/include/pcl/surface/convex_hull.h:92]: These four could be field initialized?

File: surface/include/pcl/surface/gp3.h [hunk 3]
```diff
@@ -156,26 +156,12 @@ namespace pcl
...
-        prev_is_ffn_ (false),
-        prev_is_sfn_ (false),
-        next_is_ffn_ (false),
-        next_is_sfn_ (false),
-        changed_1st_fn_ (false),
-        changed_2nd_fn_ (false),
-        new2boundary_ (),
-        already_connected_ (false)
+        eps_angle_(M_PI/4), 
+        angles_ ()
```
Comment (larshg) [inline surface/include/pcl/surface/gp3.h:163]: Is just a vector, member initialize it.

File: surface/include/pcl/surface/texture_mapping.h [hunk 4]
```diff
@@ -64,16 +64,15 @@ namespace pcl
...
+      Camera () = default;
       Eigen::Affine3f pose;
-      double focal_length;
-      double focal_length_w;  // optional
-      double focal_length_h;  // optinoal
-      double center_w;  // optional
-      double center_h;  // optional
-      double height;
-      double width;
+      double focal_length{0};
```
Comment (larshg) [inline surface/include/pcl/surface/texture_mapping.h:69]: All double values - minor nitpick.
```suggestion
      double focal_length{0.0};
```

File: tracking/include/pcl/tracking/kld_adaptive_particle_filter.h [hunk 5]
```diff
@@ -243,14 +239,14 @@ class KLDAdaptiveParticleFilterTracker
   resample() override;
 
   /** \brief the maximum number of the particles. */
-  unsigned int maximum_particle_number_;
+  unsigned int maximum_particle_number_{};
```
Comment (larshg) [inline tracking/include/pcl/tracking/kld_adaptive_particle_filter.h:242]:
```suggestion
  unsigned int maximum_particle_number_{0};
```

File: tracking/include/pcl/tracking/nearest_pair_point_cloud_coherence.h [hunk 6]
```diff
@@ -29,7 +29,7 @@ class NearestPairPointCloudCoherence : public PointCloudCoherence<PointInT> {
 
   /** \brief empty constructor */
   NearestPairPointCloudCoherence()
-  : new_target_(false), search_(), maximum_distance_(std::numeric_limits<double>::max())
+  : search_(), maximum_distance_(std::numeric_limits<double>::max())
```
Comment (larshg) [inline tracking/include/pcl/tracking/nearest_pair_point_cloud_coherence.h:32]: search_ as null_ptr and maximum distance can also be member initialized.

File: tracking/include/pcl/tracking/particle_filter.h [hunk 7]
```diff
@@ -592,14 +579,14 @@ class ParticleFilterTracker : public Tracker<PointInT, StateT> {
   std::vector<double> initial_noise_mean_;
 
   /** \brief The threshold for the particles to be re-initialized. */
-  double resample_likelihood_thr_;
+  double resample_likelihood_thr_{0.0};
 
   /** \brief The threshold for the points to be considered as occluded. */
-  double occlusion_angle_thr_;
+  double occlusion_angle_thr_{0.0};
```
Comment (larshg) [inline tracking/include/pcl/tracking/particle_filter.h:585]:
```suggestion
  double occlusion_angle_thr_{M_PI / 2.0};
```
And remove it from constructor - seems more variables could be member initialized here as well?

File: visualization/include/pcl/visualization/interactor_style.h [hunk 8]
```diff
@@ -260,36 +255,36 @@ namespace pcl
 
        protected:
         /** \brief Set to true after initialization is complete. */
-        bool init_;
+        bool init_{};
```
Comment (larshg) [inline visualization/include/pcl/visualization/interactor_style.h:258]:
```suggestion
        bool init_{false};
```

File: surface/include/pcl/surface/organized_fast_mesh.h [hunk 9]
```diff
@@ -86,20 +86,9 @@ namespace pcl
...
-      : max_edge_length_a_ (0.0f)
-      , max_edge_length_b_ (0.0f)
-      , max_edge_length_c_ (0.0f)
-      , max_edge_length_set_ (false)
-      , max_edge_length_dist_dependent_ (false)
-      , triangle_pixel_size_rows_ (1)
-      , triangle_pixel_size_columns_ (1)
-      , triangulation_type_ (QUAD_MESH)
+      : triangulation_type_ (QUAD_MESH)
       , viewpoint_ (Eigen::Vector3f::Zero ())
```
Comment (larshg) [inline surface/include/pcl/surface/organized_fast_mesh.h:90]: these last two can also be member initialized?

Comment (larshg) [reply surface/include/pcl/surface/organized_fast_mesh.h:90]: viewpoint_ and cos_angle_tolerance_

**Review by larshg — DISMISSED** [review]

File: visualization/include/pcl/visualization/mouse_event.h [hunk 10]
```diff
@@ -128,12 +128,12 @@ namespace pcl
...
-        unsigned int pointer_x_;
-        unsigned int pointer_y_;
-        unsigned int key_state_;
-        bool selection_mode_;
+        Type type_{};
+        MouseButton button_{};
+        unsigned int pointer_x_{0};
+        unsigned int pointer_y_{0};
+        unsigned int key_state_{0};
+        bool selection_mode_{false};
```
Comment (mvieth) [inline visualization/include/pcl/visualization/mouse_event.h:136]:
```suggestion
        Type type_;
        MouseButton button_;
        unsigned int pointer_x_;
        unsigned int pointer_y_;
        unsigned int key_state_{0};
        bool selection_mode_;
```
Except for `key_state_`, they are all set in the constructor based on constructor parameters

File: segmentation/include/pcl/segmentation/grabcut_segmentation.h [hunk 11]
```diff
@@ -438,39 +438,39 @@ namespace pcl
...
       /// image width
-      std::uint32_t width_;
+      std::uint32_t width_{0};
       /// image height
-      std::uint32_t height_;
+      std::uint32_t height_{0};
       // Variables used in formulas from the paper.
       /// Number of GMM components
-      std::uint32_t K_;
+      std::uint32_t K_{0};
```
Comment (mvieth) [inline segmentation/include/pcl/segmentation/grabcut_segmentation.h:446]:
```suggestion
      std::uint32_t K_;
```
Set in the constructor

File: segmentation/include/pcl/segmentation/grabcut_segmentation.h [hunk 12]
```diff
@@ -438,39 +438,39 @@ namespace pcl
...
       /// lambda = 50. This value was suggested the GrabCut paper.
-      float lambda_;
+      float lambda_{0.0f};
```
Comment (mvieth) [inline segmentation/include/pcl/segmentation/grabcut_segmentation.h:448]:
```suggestion
      float lambda_;
```
Set in the constructor

File: segmentation/include/pcl/segmentation/grabcut_segmentation.h (hunk 11 above)
Comment (larshg) [reply segmentation/include/pcl/segmentation/grabcut_segmentation.h:446]: Missed this one?

File: segmentation/include/pcl/segmentation/grabcut_segmentation.h [hunk 13]
```diff
@@ -271,7 +271,7 @@ namespace pcl
         getEpsilon () { return (epsilon_); }
         /** set epsilon which will be added to the covariance matrix diagonal which avoids singular
           * covariance matrix
-          * \param[in] epsilon user defined epsilon
+          * \param[in] epsilon user defined epsilonT
```
[the remark this answers is not in the document]
Comment (gnawme) [reply segmentation/include/pcl/segmentation/grabcut_segmentation.h]: No idea where that came frome

**Review by mvieth — APPROVED** [review]
Thank you!

**Pull request merged**